### ISSUE TITLE
Unread list filter d

### ISSUE THREAD

**Issue by haecker-felix** [element-hq/element-x-ios#3070]
### Steps to reproduce

**1. Where are you starting? What can you see?**

The iOS Element X app shows a unread badge -> 1 Unread message. 

<img src="https://github.com/user-attachments/assets/82b54949-eaa9-40c3-b0fc-3e16d914abe1" width="100">

**2. What do you click?**

Tap on the app to see the unread messages, and apply the "Unread" filter on the top. 

#### What did you expect?

Only seeing rooms with notification settings being set to
- "Standard"
- "All messages"
- or rooms where I got pinged / some of my keyword got used

Basically something like the legacy Element App is doing:

<img src="https://github.com/user-attachments/assets/6d079422-2a49-4719-9009-a99d4439905d" width="300">

#### What happened instead?

The filter (unread == "Ungelesen") shows _all_ unread rooms without respecting my notification settings. The unread badge count doesn't match the count of unread rooms. 

<img src="https://github.com/user-attachments/assets/aba0d47e-d9d3-43e6-9153-d183d5ff9e71" width="300">

Unfortunately this makes Element X unusable for me, because I'm joined in very many different rooms, and I need some view / filter for only showing the unread rooms (like the legacy Element app does). 

### Your phone model

iPhone 15 Pro

### Operating system version

iOS 18.0

### Application version

1.6.13

### Homeserver

Synapse 1.111.0 / Sliding Sync Proxy 0.99.19

### Will you send logs?

Yes

**Comment by Velin92** [comment]
Hello can we get more details? The app should be showing exactly your settings.
Considering that noisy notifications which are translated into push notifications, are not the same as "notifications", for which we intend ANY notification that updates the room list.

**Comment by haecker-felix** [comment]
> Maybe you would like the room list to only filter based on your noisy notifications?

Yes, I think that makes sense (and is also what the legacy Element app does).

**Comment by Velin92** [comment]
I see, as of right now we don't provide that option, but will convert this issue into an enhancement proposal
